### PR TITLE
Rename Pcoq into Procq and introduce a deprecated module alias.

### PR DESCRIPTION
**Pull request by ppedrot**
The Great Rocqification enters its non-trivial phase.

Overlays:
- https://github.com/ejgallego/coq-lsp/pull/865
- https://github.com/ejgallego/coq-serapi/pull/427
- https://github.com/coq-tactician/coq-tactician/pull/91

### CODE REVIEW

**Comment by ppedrot** [comment]
(I kept the two commits separates for better git blaming.)

**Review by proux01 — APPROVED** [review]

**Comment by ejgallego** [comment]
I can take care of the coq-lsp overlay folks, once you ping me this is stable.

**Comment by ejgallego** [comment]
Here it is:

https://github.com/ejgallego/coq-lsp/pull/865

It was actually trivial, but I was wary about potential issues.

**Comment by ppedrot** [comment]
@ejgallego Technically if you're willing to explicitly suppress a deprecation warning you can even make this overlay backward compatible. (That's the reason I wanted to have a deprecated alias.)

**Comment by herbelin** [comment]
Actually nothing force us to literally rename pcoq into procq. Pcoq was not very speaking already and it could also be, say, `hiparser.ml`, `paentries.ml`, `topparser.ml`, `parsertop.ml`, `parserdriver.ml`, ... or if one wants rocq in the name, `parocq.ml`, `rocqparser.ml`, ... possibly also in camel case, as in the new names we give to files these days.

File: parsing/g_constr.mlg [hunk 1]
```diff
@@ -20,9 +20,9 @@ open Constrexpr
 open Constrexpr_ops
 open Util
 
-open Pcoq
-open Pcoq.Prim
-open Pcoq.Constr
+open Procq
+open Procq.Prim
+open Procq.Constr
```
Comment (jfehrle) [inline parsing/g_constr.mlg:25]: Not to get in the way of progress, but isn't this mighty likely to break plugins that parse things?  Do we care?  Would be nice for the release notes to mention some of the things that will break when this is committed.

Comment (Zimmi48) [reply parsing/g_constr.mlg:25]: Indeed. We have a `dev/doc/changes.md` file which should be used for this kind of notes.

Comment (ppedrot) [reply parsing/g_constr.mlg:25]: The alias will keep things working, at the cost of a deprecation warning. I did check on reasonable examples.

**Comment by ppedrot** [comment]
I'm arguing against the changelog. We already have a deprecation, which is machine-checked and thus strictly better.

**Comment by Zimmi48** [comment]
I don't see what harm the changelog would do in addition to the deprecation warning. But up to you.

**Comment by proux01** [comment]
@ppedrot apparently we still need some serapi overlay and this should be good to go

**Comment by ppedrot** [comment]
@Zimmi48 I'm not a huge fan of drowning the important high-level changes of the OCaml API into minutiae about the precise renaming of some functions when the latter is already checked through warnings.

**Comment by ppedrot** [comment]
@proux01 I screwed the overlay in the previous iteration, this should be fixed now hopefully.

**Comment by proux01** [comment]
@coqbot merge now

**Comment by ejgallego** [comment]
> @ejgallego Technically if you're willing to explicitly suppress a deprecation warning you can even make this overlay backward compatible. (That's the reason I wanted to have a deprecated alias.)

We follow a 0-warning policy via our inherited JS coding rules. Indeed, there is little point on keeping the deprecated call in cases like this where the fix is easy. You just push the burden to the moment the deprecated API has been removed, and thus strictly increase the complexity of your setup for that time.

**Pull request merged**